src/kafka/connection/kafka_connection.hh

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#pragma once
+
+#include "tcp_connection.hh"
+#include "../protocol/headers.hh"
+#include "../protocol/api_versions_request.hh"
+#include "../protocol/api_versions_response.hh"
+
+#include <boost/iostreams/device/back_inserter.hpp>
+#include <boost/iostreams/stream.hpp>
+#include <boost/iostreams/device/array.hpp>
+
+namespace seastar {
+
+namespace kafka {
+
+class kafka_connection {
+private:
+    lw_shared_ptr<tcp_connection> _connection;
+    std::string _client_id;
+    int32_t _correlation_id;
+    api_versions_response _api_versions;
+
+    template<typename RequestType>
+    future<int32_t> send_request(const RequestType &request, int16_t api_version) {
```

Comment:
Same as above, as much as I also prefer the way of "type &name", they use the other one
```suggestion
    future<int32_t> send_request(const RequestType& request, int16_t api_version) {
```